### DISCUSSION TITLE
Loading router in packages

### DISCUSSION THREAD

**Discussion by rafael-franca** [remix-run/remix#4960] [General]
What do you think about improve the router to accept that package creators can append routers, and/or extends paths to te router?

Eg.
I am developing a blogging system. I have this routes:

```
/
/$slug
```

I want to append the admin routes that have in external, to this:

```
/
/$slug
/admin/... -> /node_modules/admin-package/routes/...
```

So, when try access any route starting with `/admin`, will be loading the routes in package.

**Comment by sergiodxa** [comment]
You can use the `remix.config.js` routes function to add more routes from a package <https://remix.run/docs/en/v1/file-conventions/remix-config#routes>, your `admin-package` could expose a function that receives the `route` function and any other extra configuration and sets the routes for you on your app.
